### PR TITLE
fixed solving generic type of JavassistFieldDeclaration

### PR DESCRIPTION
**Pull request by mlangkabel**

### CODE REVIEW

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java [hunk 1]
```diff
@@ -41,9 +44,16 @@ public JavassistFieldDeclaration(CtField ctField, TypeSolver typeSolver) {
     @Override
     public Type getType() {
         try {
-            return JavassistFactory.typeUsageFor(ctField.getType(), typeSolver);
+            if (ctField.getGenericSignature() != null && declaringType() instanceof TypeParametrizable) {
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java]: Why do we need the second condition? Can a field be declared in a type that is not parametrizable (an enum maybe?) If that is the case the field could not be generic, right?

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java [hunk 2]
```diff
@@ -51,8 +51,8 @@
                 }
                 try {
                     if (method.getGenericSignature() != null) {
-                        SignatureAttribute.MethodSignature classSignature = SignatureAttribute.toMethodSignature(method.getGenericSignature());
-                        List<Type> parametersOfReturnType = parseTypeParameters(classSignature.getReturnType().toString(), typeSolver, invokationContext);
+                        SignatureAttribute.MethodSignature methodSignature = SignatureAttribute.toMethodSignature(method.getGenericSignature());
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: right

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java (hunk 1 above)
Comment (mlangkabel) [reply java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java]: It's just to make sure that the following cast would work. I like to do the instanceof test before actually casting to a different type. So I don't want this code to break if someone changes something in the rest of the code base.

Comment (ftomassetti) [reply java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java]: Ok, I think I should change declaringType to return a ReferenceTypeDeclaration and then remove this check. For now I will merge this as it is.

**Pull request merged**